junit-jupiter-params/src/test/java/org/junit/jupiter/params/provider/CsvArgumentsProviderTests.java

```diff
@@ -143,6 +143,33 @@ void defaultEmptyValueAndDefaultNullValue() {
 		assertThat(arguments).containsExactly(array("", "null", null, "apple"));
 	}
 
+	@Test
+	void defaultLeadingSpaces() {
+		var annotation = csvSource("   as, null, , apple");
+
+		var arguments = provideArguments(annotation);
+
+		assertThat(arguments).containsExactly(array("as", "null", null, "apple"));
+	}
+
+	@Test
+	void defaultTrailingSpaces() {
+		var annotation = csvSource("as      , null, , apple");
+
+		var arguments = provideArguments(annotation);
+
+		assertThat(arguments).containsExactly(array("as", "null", null, "apple"));
+	}
+
+	@Test
+	void defaultLeadingAndTrailingSpaces() {
+		var annotation = csvSource("        as      , null, , apple");
+
+		var arguments = provideArguments(annotation);
+
+		assertThat(arguments).containsExactly(array("as", "null", null, "apple"));
+	}
+
```

Comment:
Please also add test for when the new annotation attribute is set to `false`.

Reply:
added, see the latest commit.